coqpit/coqpit.py

```diff
@@ -608,56 +643,46 @@ def load_json(self, file_name: str) -> None:
         self = self.deserialize(dump_dict)  # pylint: disable=self-cls-assignment
         self.check_values()
 
-    def parse_args(self, args: argparse.Namespace) -> None:
+    def parse_args(self, args: Optional[argparse.Namespace] = None) -> None:
         """Update config values from argparse arguments with some meta-programming ✨.
 
         Args:
-            args (namespace): argeparse namespace as an output of ```argparse.parse_arguments()```.
+            args (namespace, optional): argeparse namespace as an output of ```argparse.parse_arguments()```. If unspecified will use a newly created parser with ```init_argparse()```.
 
         Returns:
             Coqpit: new config object with updated values.
         """
+        if not args:
+            parser = self.init_argparse()
+            args = parser.parse_args()
         if isinstance(args, argparse.Namespace):
             args_dict = vars(args)
         elif isinstance(args, list):
             # overriding values from .parse_known_args()
             args_dict = dict([*zip(args[::2], args[1::2])])
```

Comment:
One issue is there when you give a list as input. For instance, the output of `parser.parse_known_args()` is a list and `--` needs to be replaced as in the old code.

Reply:
I don't see how that's connected to this PR?

Reply:
Oh, nevermind. It's just a different change. I get it.